### ISSUE TITLE
Blog Posts Block not displaying posts correctly on front end

### ISSUE THREAD

**Issue by TheSteveK** [Automattic/wp-calypso#50672]
**Workaround: Manually set a post excerpt for the page where the block is being added. This will use the excerpt in the OG tags and will not trigger the problem.**

**This issue is now closed as a fix was merged. The fix will ship to all Atomic sites in the next version of Jetpack (release date is still TBD).**

*********

We have found several instances of Atomic sites where a page has the blog posts block.

---

In the editor the block displays correctly and shows the expected posts.  When previewing the page or viewing the published page on the live site, the block displays unexpected behavior.  

A few different symptoms have been reported including

- no posts shown at all
- only a selection of older posts show
- Some show most posts while skipping 

### Steps to reproduce the behavior

This is proving difficult to consistently reproduce.

1. Onn a selected site open the blog page in the editor
2. Observe the blog posts block and the posts displayed
3. Preview the page
4. Notice the difference in the posts displayed if any display at all.

@Gilmoursa mentions that the post block seems to work okay if the page is set as the homepage but not on other pages.

The impacted sites so far have all been wp.com atomic sites, no simple site examples have been found.

Original internal thread with examples: p1614708782171000-slack-C03TY6J1A.  
New triaging thread: p1614716941027400-slack-C02FMH4G8

@dkoo requested assignment.

**Comment by TheSteveK** [comment]
Example issue: #3786830-zen

**Comment by Gilmoursa** [comment]
Posts display on the homepage using the blog post block, but do not display on separate pages, such as a blog page, using the blog posts block.

**Comment by Gilmoursa** [comment]
For anyone coming to this issue looking for a temporary solution, as a workaround, you can add the 'latest posts' block.

Link: https://wordpress.com/support/wordpress-editor/blocks/latest-posts-block/

**Comment by happychait** [comment]
Got one in 28133063-hc

**Comment by mikeicode** [comment]
Another 3787525-zd-woothemes

**Comment by Gilmoursa** [comment]
Another 22599439-hc

**Comment by ktyfuller604** [comment]
Another report here 3787071-zen

**Comment by tbradsha** [comment]
27285571-hc

**Comment by JoshuaGoode** [comment]
Updated original comment to include new internal triaging thread: p1614716941027400-slack-C02FMH4G8

**Comment by tbradsha** [comment]
Out of order posts reported in 3787588-zen using the Blog Posts block. WordPress.com Atomic.

**Comment by benchilcote** [comment]
Another case here: 26970422-hc / 3787617-zen

**Comment by wprhunt** [comment]
Howdy, got another here:
27877583-hc / 3787899-zen

**Comment by thesacredpath** [comment]
Another one here: 3787476-zen / 28100796-hc

**Comment by carladoria** [comment]
another one 27821778-hc / 3788021-zen

**Comment by JoshuaGoode** [comment]
Discovered that deactivating Jetpack's `publicize` and `sharedaddy` modules can restore the block back to working order.

**Comment by tavilla61** [comment]
Another one:  28136657-hc / 3788030-zen

**Comment by dkoo** [comment]
Reassigning to @apeatling since this appears to be isolated to Atomic + WP.com environments. Please feel free to bring me back in if a resolution might require anything on the Newspack Blocks side.

**Comment by sbathompson-he** [comment]
Saw on these two chats too, they are on the ticket now. Zen - 3786938 & 3787936

**Comment by jamiepalatnik** [comment]
Another report: 28072414-hc
Deactivating Sharing and Publicize modules resolved the issue as a temporary workaround.

**Comment by joweber123** [comment]
Another one: 27749652-hc / 3788209-zen

**Comment by jerrysarcastic** [comment]
Also reported in https://wordpress.com/forums/topic/theme-not-displaying-all-posts/

I let the user know about the sharing/publicize settings workaround and will follow up once this issue is closed.

**Comment by gwensmithx** [comment]
Also reported here #3788229-zd / #28138200-hc

I've turned off sharing and that worked for now.

**Comment by devNigel** [comment]
Another report: 19914821-hc

Suggested them to keep sharing and publicize features disabled as a workaround.

**Comment by JoshuaGoode** [comment]
After extensive review, we narrowed it down to publicize and sharedaddy modules conflicting with the block.

@leogermani then identified https://github.com/Automattic/jetpack/pull/18654 as being the source of the issue which landed with Jetpack 9.5.

Potential PR for a fix: https://github.com/Automattic/jetpack/pull/18987

**Comment by leogermani** [comment]
Potential workaround (it would be great if someone could confirm it): Manually set a post excerpt for the page where the block is being added. This will use the excerpt in the OG tags and will not trigger the problem.

**Comment by yashitamittal11** [comment]
Another one- #3788484-zen

**Comment by druesome** [comment]
#28140411-hc

**Comment by pauloeaquino** [comment]
#28139941-hc

**Comment by tarunvijwani** [comment]
Another one here #3788538-zd / #28140771-hc

**Comment by rinazrina** [comment]
Also reported in: 
- #27099768-hc 
- #28140261-hc

**Comment by erricgunawan** [comment]
3787467-zen

**Comment by kriskorn** [comment]
22676002-hc

**Comment by yashitamittal11** [comment]
#3788744-zen

**Comment by kriskorn** [comment]
28144240-hc

**Comment by Robertght** [comment]
Another one in #28144913-hc

**Comment by devNigel** [comment]
Another one: 28135224-hc

**Comment by dolgelukkig** [comment]
Another one #27943142-hc follow-up ticket here #3788965-zen

**Comment by jeherve** [comment]
Everyone: as you receive more reports from customers, **could you ask them to try this work-around**, and see if that fixes the problem for them?

> Potential workaround (it would be great if someone could confirm it): Manually set a post excerpt for the page where the block is being added. This will use the excerpt in the OG tags and will not trigger the problem.

https://github.com/Automattic/wp-calypso/issues/50672#issuecomment-789375532

**Comment by kriskorn** [comment]
28148080-hc

**Comment by jartes** [comment]
@jeherve Tested the suggested workaround on one interaction and I can confirm that it worked. 

28140710-hc | Follow-up 3789141-zen

**Comment by mriyazuddin** [comment]
Another one here. The workaround did work.

17383059-hc | Follow-up: 3788463-zen

**Comment by JoshuaGoode** [comment]
Reopening the issue for searching and troubleshooting purposes until we see the Jetpack update deployed to Atomic.

**Comment by jartes** [comment]
Another one on 28153283-hc. Adding the excerpt on the page worked.

**Comment by xpurichan** [comment]
Another encounter on 28123608-hc. Follow up is 3789783-zd.

Interestingly they claimed that the blog posts block loaded properly on their "Start" (home) page, but not on a separate "Blog" page they had created. I didn't see this myself.

**Comment by Thelmachido** [comment]
Another encounter on 153666-hc. Follow up is 3789790-zd.

**Comment by Gilmoursa** [comment]
> Interestingly they claimed that the blog posts block loaded properly on their "Start" (home) page, but not on a separate "Blog" page they had created. I didn't see this myself.

I saw a similar situation in 22599439-hc

**Comment by syds87** [comment]
Similar situation on 3789681-zd and 3789273-zd.

Tested the suggested workaround and it worked for both of them.

**Comment by jartes** [comment]
Another one on 28137681-hc. 

Their theme didn't support excerpts on pages, so added excerpt support by adding

`add_post_type_support( 'page', 'excerpt' );` 

via the Code Snippets plugin.

**Comment by sanjeev00733** [comment]
Another one in #3788720-zen

**Comment by metabreakr** [comment]
28154698-hc with a follow-up in 3789964-zen

**Comment by xpurichan** [comment]
Reported in 3790103-zen. Provided the Publicize/Sharing module deactivation workaround.

**Comment by ezebecke** [comment]
27660928-hc Follow up created in 3790369-zd

**Comment by mikeicode** [comment]
28158915-hc shutting off the sharing worked. They would like a response when it's resolved so they can activate those again. 3790562-zd-woothemes

**Comment by JoshuaGoode** [comment]
Hi folks! Thanks for your continued reports.  We're waiting on the fix to be deployed in a Jetpack release.

In the meantime, let's stop deactivating the Jetpack modules and, instead, ask the user to use an Excerpt to the page(s) that are impacted by this issue.

> Manually set a post excerpt for the page where the block is being added. This will use the excerpt in the OG tags and will not trigger the problem.

**Comment by edequalsawesome** [comment]
Another report in 23662876-hc -- let them know we'd follow-up with them when the issue was resolved in 3791190-zen

**Comment by devNigel** [comment]
Another report: 

* 28166191-hc
* 27574759-hc

Suggested adding an excerpt for the page.

**Comment by sanjeev00733** [comment]
Another report in #24336406-hc. Adding excerpt to the page fixed the issue.

**Comment by nagpai** [comment]
#28103426-hc

**Comment by kavyagokul** [comment]
Also 28167995-hc

**Comment by donalirl** [comment]
#3790605-zen

**Comment by nagpai** [comment]
#28170460-hc - Atomic site - excerpt added

**Comment by javierarce** [comment]
Another #3789096-zen. The workaround solved the issue.

**Comment by AtrumGeost** [comment]
Reported in 28049316-hc

The workaround worked! Thank you!

**Comment by pauljacobson** [comment]
3787228-zen - The workaround worked here too.

**Comment by kraftbj** [comment]
The fix is being rolled out to WoA sites now. p9o2xV-1hK-p2

**Comment by jeherve** [comment]
The fix has now been deployed to all Atomic sites. I'll close this issue now.
